File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    answer = 0
    
    check = ["aya", "ye", "woo", "ma"]
    
    for word in babbling:
        flag = True
        for index, item in enumerate(check):
            word = word.replace(item, str(index))
            
            if str(index) * 2 in word:
                flag = False
                break
        
        if flag and word.isdigit(): 
            answer += 1

    return answer

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js


function solution(babbling) {
    const words = ["aya", "ye", "woo", "ma"];
    
    let answer = 0;
    for (let i = 0; i < babbling.length; i++) {
        let flag = true;
        let targetItem = babbling[i];
        for (const word of words) {
            if (targetItem.includes(word.repeat(2))) {
                flag = false;
                break;
            }
        }
        
        if (flag) {
            words.forEach((word, index) => {
                targetItem = targetItem.replaceAll(word, index);
            })
            if (!isNaN(targetItem)) answer++;

        }
    }
    
    return answer;
}